contracts/account/signers/SignerRSA.sol

```diff
@@ -3,12 +3,14 @@
 pragma solidity ^0.8.20;
 
 import {RSA} from "../../utils/cryptography/RSA.sol";
-import {Clones} from "../../proxy/Clones.sol";
 import {EIP712ReadableSigner} from "./EIP712ReadableSigner.sol";
 
 abstract contract SignerRSA is EIP712ReadableSigner {
+    bytes private _e;
+    bytes private _n;
+
     function signer() public view virtual returns (bytes memory e, bytes memory n) {
-        return abi.decode(Clones.fetchCloneArgs(address(this)), (bytes, bytes));
+        return (_e, _n);
```

Comment:
This is going in storage. That is not complying with 4337 storage accesses.

Reply:
Right, even in immutable arguments it accesses storage via EIP712Readable. It also discards upgradeability and most other account providers I've reviewed don't identify this as an issue or just don't care about it.

I asked about this in [Alto (Pimlico)](https://github.com/pimlicolabs/alto/issues/260) and it seems like these transactions will be processed privately but not broadcasted to the decentralized mempool (whatever that is). So far, Accounts can read their own storage or external "associated storage" (i.e. `keccak(account_address||bytes32)+n` where `n` is up to 128) so that enables to validate multiple contract signatures using `IERC1271` through a validator.

With that approach, a MultisigValidator is just a multiplexer of multiple accounts checked with `SignatureChecker.isValidSignatureNow` and is still compatible during the validation phase. However, any upgradeable version of the account or associated modules breaks the rules again (checks non-associated storage when reading the implementation slot).

From what I've seen (e.g. Safe), Multisigs frequently work by collecting the signatures and then packing them into a transaction that someone else's execute, for that reason I think it's fine that multisig-approved calls are not validated during the execution phase, and rather executed by another account or relayer.

Bottom line is: Our default recommendation is an `AccountERC7579` that requires at least 1 validator module. Validator modules shouldn't be upgradeable or they'll break storage access rules although the account can still be executed from another account or through a private bundler (e.g Pimlico whitelists Safe proxy accounts even though they read the master copy from storage).